File: database_sample/create_excel.py
"""
PilateStudio Database Excel Creator
Creates a single Excel file with multiple sheets from CSV files
"""

import pandas as pd
import os
from pathlib import Path

def create_excel_database():
    """Create Excel file from CSV files"""
    
    # Get current directory
    current_dir = Path(__file__).parent
    
    # Output Excel file
    excel_file = current_dir / 'PilateStudio_Database.xlsx'
    
    # CSV files to process
    csv_files = {
        'Users': 'Users.csv',
        'Trainers': 'Trainers.csv',
        'Classes': 'Classes.csv',
        'Schedules': 'Schedules.csv',
        'Bookings': 'Bookings.csv',
        'Transactions': 'Transactions.csv'
    }
    
    print("🔧 Creating Excel file from CSV data...")
    print("-" * 50)
    
    try:
        # Create Excel writer
        with pd.ExcelWriter(excel_file, engine='openpyxl') as writer:
            for sheet_name, csv_filename in csv_files.items():
                csv_path = current_dir / csv_filename
                
                if not csv_path.exists():
                    print(f"⚠️  Warning: {csv_filename} not found, skipping...")
                    continue
                
                # Read CSV file
                df = pd.read_csv(csv_path)
                
                # Write to Excel sheet
                df.to_excel(writer, sheet_name=sheet_name, index=False)
                
                # Format the sheet
                worksheet = writer.sheets[sheet_name]
                
                # Auto-adjust column widths
                for column in worksheet.columns:
                    max_length = 0
                    column_letter = column[0].column_letter
                    
                    for cell in column:
                        try:
                            if len(str(cell.value)) > max_length:
                                max_length = len(str(cell.value))
                        except:
                            pass
                    
                    adjusted_width = min(max_length + 2, 50)
                    worksheet.column_dimensions[column_letter].width = adjusted_width
                
                # Make header row bold
                for cell in worksheet[1]:
                    cell.font = cell.font.copy(bold=True)
                
                print(f"✓ Sheet created: {sheet_name} ({len(df)} rows)")
        
        print("-" * 50)
        print(f"✅ Success! Excel file created:")
        print(f"   📁 {excel_file}")
        print(f"\n📊 Summary:")
        print(f"   • 6 sheets created")
        print(f"   • Ready to upload to Google Sheets")
        print(f"   • Or use directly in Excel")
        
        return True
        
    except ImportError:
        print("\n❌ Error: Required packages not installed")
        print("\n📦 Please install required packages:")
        print("   pip install pandas openpyxl")
        return False
        
    except Exception as e:
        print(f"\n❌ Error creating Excel file: {str(e)}")
        return False

if __name__ == "__main__":
    print("=" * 50)
    print("  PilateStudio Database - Excel Creator")
    print("=" * 50)
    print()
    
    success = create_excel_database()
    
    if success:
        print("\n🚀 Next steps:")
        print("   1. Open PilateStudio_Database.xlsx")
        print("   2. Verify all data is correct")
        print("   3. Upload to Google Drive")
        print("   4. Open with Google Sheets")
        print("   5. Share with service account")
        print("   6. Copy spreadsheet ID to .env")
    else:
        print("\n💡 Alternative:")
        print("   Import CSV files directly to Google Sheets")
        print("   See README.md for instructions")
    
    print()

File: database_sample/create_excel.js
/**
 * PilateStudio Database Excel Creator
 * Creates a single Excel file with multiple sheets from CSV files
 * 
 * Usage: node create_excel.js
 */

const fs = require('fs');
const path = require('path');

// Simple CSV to Array converter
function csvToArray(csvText) {
  const lines = csvText.trim().split('\n');
  return lines.map(line => {
    // Simple CSV parsing (handles basic comma separation)
    return line.split(',').map(cell => cell.trim());
  });
}

// Create Excel-compatible HTML
function createExcelHTML() {
  const csvFiles = [
    'Users.csv',
    'Trainers.csv',
    'Classes.csv',
    'Schedules.csv',
    'Bookings.csv',
    'Transactions.csv'
  ];

  let htmlContent = `
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>PilateStudio Database</title>
  <style>
    body { font-family: Arial, sans-serif; padding: 20px; }
    h1 { color: #4a3829; }
    h2 { color: #8b6f47; margin-top: 30px; }
    table { border-collapse: collapse; width: 100%; margin-bottom: 30px; }
    th { background-color: #d4c5b0; font-weight: bold; padding: 10px; border: 1px solid #999; }
    td { padding: 8px; border: 1px solid #ddd; }
    tr:nth-child(even) { background-color: #f9f9f9; }
    .info { background: #e8f5e9; padding: 15px; border-left: 4px solid #4caf50; margin-bottom: 20px; }
  </style>
</head>
<body>
  <h1>📊 PilateStudio Database</h1>
  <div class="info">
    <strong>Instructions:</strong><br>
    1. Open this file in Excel/LibreOffice<br>
    2. Or copy tables to Google Sheets<br>
    3. Share with service account: pilatestudiostella@pilatestudio-stella.iam.gserviceaccount.com
  </div>
`;

  csvFiles.forEach(csvFile => {
    const csvPath = path.join(__dirname, csvFile);
    
    if (!fs.existsSync(csvPath)) {
      console.log(`⚠️  Warning: ${csvFile} not found, skipping...`);
      return;
    }

    const csvContent = fs.readFileSync(csvPath, 'utf-8');
    const rows = csvToArray(csvContent);
    
    const sheetName = csvFile.replace('.csv', '');
    
    htmlContent += `\n  <h2>${sheetName} (${rows.length - 1} rows)</h2>\n`;
    htmlContent += `  <table>\n`;
    
    // Header row
    htmlContent += `    <tr>\n`;
    rows[0].forEach(header => {
      htmlContent += `      <th>${header}</th>\n`;
    });
    htmlContent += `    </tr>\n`;
    
    // Data rows
    for (let i = 1; i < rows.length; i++) {
      htmlContent += `    <tr>\n`;
      rows[i].forEach(cell => {
        htmlContent += `      <td>${cell}</td>\n`;
      });
      htmlContent += `    </tr>\n`;
    }
    
    htmlContent += `  </table>\n`;
    
    console.log(`✓ Processed: ${sheetName} (${rows.length - 1} rows)`);
  });

  htmlContent += `
</body>
</html>`;

  return htmlContent;
}

// Main function
function main() {
  console.log('=' .repeat(50));
  console.log('  PilateStudio Database - HTML/Excel Creator');
  console.log('=' .repeat(50));
  console.log();
  console.log('🔧 Creating HTML file from CSV data...');
  console.log('-'.repeat(50));

  try {
    const htmlContent = createExcelHTML();
    const outputFile = path.join(__dirname, 'PilateStudio_Database.html');
    
    fs.writeFileSync(outputFile, htmlContent, 'utf-8');
    
    console.log('-'.repeat(50));
    console.log('✅ Success! HTML file created:');
    console.log(`   📁 ${outputFile}`);
    console.log();
    console.log('📊 How to use:');
    console.log('   1. Open PilateStudio_Database.html in browser');
    console.log('   2. Copy each table to Google Sheets');
    console.log('   3. Or open in Excel: File > Open > Select .html file');
    console.log();
    console.log('🚀 Alternative:');
    console.log('   Import CSV files directly to Google Sheets');
    console.log('   (See PETUNJUK_IMPORT.md for instructions)');
    console.log();
    
  } catch (error) {
    console.error('❌ Error:', error.message);
    return false;
  }
}

// Run
main();
